tests/test_cms_menus.py

```diff
@@ -1495,3 +1543,188 @@ def test_menu_with_multiple_languages(self):
+        self.assertNotIn(not_main_navigation, result)
+
+    def test_main_navigation_not_found(self):
+        """
+        When none of the MenuContent objects are related to a Menu marked as the main navigation, the queryset is
+        returned unchanged
+        """
+        factories.MenuContentFactory.create_batch(2, language="en", menu__main_navigation=False)
+        all_menucontents = MenuContent._base_manager.all()
+
+        result = self.menu.get_main_navigation(menucontents=all_menucontents)
+
+        self.assertEqual(result, all_menucontents)
+        self.assertEqual(all_menucontents.count(), result.count())
+
+    def test_multiple_main_navigation(self):
+        """
+        When multiple MenuContent objects are related to a Menu marked as the main navigation are found, all the
+        MenuContent objects should be returned
+        """
+        older = factories.MenuContentWithVersionFactory(language="en", menu__main_navigation=True)
+        newer = factories.MenuContentWithVersionFactory(language="en", menu__main_navigation=True)
+        all_menucontents = MenuContent._base_manager.all()
+
+        result = self.menu.get_main_navigation(menucontents=all_menucontents)
+
+        self.assertEqual(result.count(), 2)
+        self.assertIn(newer, result)
+        self.assertIn(older, result)
+
+
+@override_settings(MAIN_NAVIGATION_ENABLED=True)
+class MainNavigationIntegrationTestCase(CMSTestCase):
+    """
+    Integration tests to check which navigation menu is displayed when having multiple Menu and MenuContent objects and
+    going through the steps to mark one as the main navigation
+    """
+    def setUp(self):
+        # create a page and urls we can use to make a request with the test client
+        self.pagecontent = factories.PageContentWithVersionFactory(
+            language="en",
+            version__created_by=self.get_superuser(),
+            version__state=PUBLISHED,
+        )
+        self.live_url = self.pagecontent.get_absolute_url()
+        self.edit_url = get_object_edit_url(self.pagecontent, language="en")
+        self.preview_url = get_object_preview_url(self.pagecontent, language="en")
+        # create two menus, neither marked as the main navigation
+        self.first_menucontent = factories.MenuContentWithVersionFactory(
+            language="en",
+            version__state=PUBLISHED,
+            menu__main_navigation=False,
+            menu__site_id=1,
+        )
+        self.first_menucontent_child = factories.ChildMenuItemFactory(parent=self.first_menucontent.root)
+        self.second_menucontent = factories.MenuContentWithVersionFactory(
+            language="en",
+            version__state=PUBLISHED,
+            menu__main_navigation=False,
+            menu__site_id=1,
+        )
+        self.second_menucontent_child = factories.ChildMenuItemFactory(parent=self.second_menucontent.root)
+
+    def _get_nav_from_response(self, response):
```

Comment:
These helpers may be useful for other tests in the future, could you add this to a utils file or something in the test_utils directory?